docs/05_best-practices/08_abi/02_manually_write_an_ABI_file_explained.md

```diff
@@ -0,0 +1,11 @@
+## Manually write, or fix, an ABI file
+- Advanced features of the newest version of the ABI will require manual construction of the ABI, and odd and advanced C++ patterns could capsize the generator's type deductions. So having a good knowledge of how to write an ABI should be an essential piece of knowledge of a smart contract writer.
+- Please refer to [developers.eos.io "How to Write an ABI File"](https://developers.eos.io/eosio-cpp/docs/how-to-write-an-abi) to learn about the different sections of an ABI.
+
+### Adding Ricardian Contracts and Clauses to ABI
+- , the ABI generator will try to automatically import contracts and clauses into the generated ABI.  There are a few caveats to this, one is a strict naming policy of the files and an HTML tag used to mark each Ricardian contract and each clause.
+- The Ricardian contracts should be housed in a file with the name <contract name>.contracts.md and the clauses should be in a file named <contract name>.clauses.md.
```

Comment:
The files should be styled so they stand out, either with an inline-code (backticks) or bold.